### PR TITLE
change the delete and put so that both are idempotent

### PR DESCRIPTION
**Pull request by p2hang**
delete always return 204

and put changes the quantity of a product to certain amount instead of add/remove on the existing quantity

### CODE REVIEW

File: redis_inventory.py [hunk 1]
```diff
@@ -58,12 +58,7 @@ def put_product(self, id, info):
 
 
   def delete_product(self, product_id):
-    if self.redis.exists(product_id):
-      self.redis.delete(product_id)
-      return True
-    else:
-      return False
-
+    self.redis.delete(product_id)
```
Comment (senongtor) [inline redis_inventory.py]: Why did you delete the existence check..? It make sense to remove return value but I think we need to keep existence check just in case that customer tries to delete non-existing things. @plzhang

File: server.py [hunk 2]
```diff
@@ -75,8 +75,8 @@ def get_one_product(id):
 
 @app.route('/inventory/products/<int:id>', methods=['PUT'])
 def update_to_product(id):
-  """ add certain amount to product
-  This method will add certain amount to product in the inventory
+  """ change product quantity to certain amount
+  This method will product change product quantity to certain amount in the inventory
```
Comment (senongtor) [inline server.py]: extra word product.. @plzhang

File: redis_inventory.py (hunk 1 above)
Comment (p2hang) [reply redis_inventory.py]: Thanks! From the lecture, Delete should only return 204 No Content even if the resource wasn't found because Delete is ***idempotent***.

File: server.py (hunk 2 above)
Comment (p2hang) [reply server.py]: Thanks fixed!

File: redis_inventory.py (hunk 1 above)
Comment (p2hang) [reply redis_inventory.py]: Will check if the product_id exists.
